blr.py

```diff
@@ -1,42 +1,44 @@
+from functools import partial
+
 import numpy as np
 import matplotlib.pyplot as plt
 
-#data generate: sin(2pix)
-#BLR
-
 
 class BLR:
-    def __init__(self, alpha, beta, dimension):
+    def __init__(self, alpha: float, beta: float, dimension: int):
```

Comment:
By providing type hinting, it is possible to provide easier use to third parties.

See also [python doc for typing](https://docs.python.org/3/library/typing.html).